dlib/dnn/cpu_dlib.cpp

```diff
@@ -1699,6 +1699,28 @@ namespace dlib
             }
         }
 
+        void tensor_conv::forward_without_setup(
```

Comment:
Nope not quiet it doesn't set

last_stride_y = stride_y;
last_stride_x = stride_x;
last_padding_y = padding_y;
last_padding_x = padding_x;

 that's the only "setup" the CPU implementation seems to do.

You could call this one in the other one to make it nicer and reduce code duplication yes. 
I thought about that but left it out for now.

The reason why is there is a much larger difference in the CUDA version but I have to keep the same signatures as interface between the two have to stay the same.